src/paths.md

```diff
@@ -103,3 +92,79 @@ mod a {
 }
 # fn main() {}
 ```
+
+## Canonical paths
+
+Items defined in a module or implementation have a *canonical path* that
+corresponds to where within its crate it is defined. All other paths to these
+items are aliases. The canonical path is defined as a *path prefix* appended by
+the path component the item itself defines.
+
+[Implementations] and [use declarations] do not have canonical paths, although
+the associated items that implementations define do have them. Items defined in
```

Comment:
i wonder if "associated items" is too overloaded a term here, it made me do a double-take at first.

Reply:
It's technically extraneous, as implementations only define associated items. I removed it; does it flow better that way?